lib/checklists/action.rb

```diff
@@ -1,44 +1,38 @@
 class Checklists::Action
+  include ActiveModel::Validations
+
   CONFIG_PATH = Rails.root.join('lib', 'checklists', 'actions.yaml')
 
-  attr_accessor :id,
-                :title,
-                :consequence,
-                :exception,
-                :title_url,
-                :lead_time,
-                :criteria,
-                :audience,
-                :guidance_link_text,
-                :guidance_url,
-                :guidance_prompt,
-                :priority
-
-  def initialize(params)
-    @id = params['action_id']
-    @title = params['title']
-    @consequence = params['consequence']
-    @exception = params['exception']
-    @title_url = params['title_url']
-    @lead_time = params['lead_time']
-    @criteria = params['criteria']
-    @audience = params['audience']
-    @guidance_link_text = params['guidance_link_text']
-    @guidance_url = params['guidance_url']
-    @guidance_prompt = params['guidance_prompt']
-    @priority = params['priority']
+  validates_presence_of :id, :title, :consequence, :criteria
+  validates_inclusion_of :audience, in: %w(business citizen)
+  validates_presence_of :guidance_link_text, if: :guidance_url
+  validates_format_of :priority, with: /\d+/
```

Comment:
Would be nice to do this:
```suggestion
  validates_numericality_of :priority, : only_integer
```
But I assume priority is being parsed as a string from the YAML?

Reply:
Oh yeah - done 👍